### PR TITLE
mariadb 10.2.6, mariadb@10.1 (new formula)

### PR DESCRIPTION
**Pull request by makigumo**
* update to stable 10.2.6 (https://mariadb.com/kb/en/mariadb/mariadb-1026-release-notes/)
* move 10.1.23 to new mariadb@10.1
* remove mariadb@10.1 alias
* create mariadb@10.2 alias

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/mariadb@10.1.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  desc "Drop-in replacement for MySQL"
+  homepage "https://mariadb.org/"
+  url "https://ftp.osuosl.org/pub/mariadb/mariadb-10.1.23/source/mariadb-10.1.23.tar.gz"
+  sha256 "54d8114e24bfa5e3ebdc7d69e071ad1471912847ea481b227d204f9d644300bf"
+
+  bottle do
+    sha256 "04cd62eea7f3209cbb84c3f0ae272a1506895fe694adbde590c2cdf070fa4de5" => :sierra
+    sha256 "fac8cb5ab929041a7b29b28f18d1f1f0e5fcef1b197fef330f3f4cc783529532" => :el_capitan
+    sha256 "a0dc238797384b8bb9e72b7c4bd5f7faa37d16fab61676fb01c776c765fea061" => :yosemite
+  end
```
Comment (JCount) [inline Formula/mariadb@10.1.rb]: Please don't include the bottle block when creating a new formula.

File: Formula/mariadb@10.1.rb [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
```
Comment (JCount) [inline Formula/mariadb@10.1.rb]: this formula needs to include the line `keg_only :versioned_formula`

File: Formula/mariadb.rb [hunk 3]
```diff
@@ -39,14 +34,6 @@ class Mariadb < Formula
     :because => "both install plugins"
 
   def install
-    if build.devel?
```
Comment (ilovezfs) [inline Formula/mariadb.rb]: I'm a little confused. I would think this section would now apply to 10.2.6 stable if it applied to 10.2.5 when 10.2.5 was devel.

File: Formula/mariadb.rb [hunk 4]
```diff
@@ -114,13 +98,7 @@ def install
     bin.install_symlink prefix/"scripts/mysql_install_db"
 
     # Fix up the control script and link into bin
-    inreplace "#{prefix}/support-files/mysql.server" do |s|
-      s.gsub!(/^(PATH=".*)(")/, "\\1:#{HOMEBREW_PREFIX}/bin\\2")
-      if build.devel?
-        # pidof can be replaced with pgrep from proctools on Mountain Lion
-        s.gsub!(/pidof/, "pgrep") if MacOS.version >= :mountain_lion
```
Comment (ilovezfs) [inline Formula/mariadb.rb]: same comment here

File: Formula/mariadb@10.1.rb [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    system "cmake", ".", *std_cmake_args, *args
+    system "make"
+    system "make", "install"
+
+    # Fix my.cnf to point to #{etc} instead of /etc
+    (etc/"my.cnf.d").mkpath
+    inreplace "#{etc}/my.cnf", "!includedir /etc/my.cnf.d",
+                               "!includedir #{etc}/my.cnf.d"
+    touch etc/"my.cnf.d/.homebrew_dont_prune_me"
```
Comment (JCount) [inline Formula/mariadb@10.1.rb]: Currently, this will conflict with the main formula, is that an issue?

File: Formula/mariadb@10.1.rb [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/mysqld_safe</string>
+        <string>--datadir=#{var}/mysql</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+      <key>WorkingDirectory</key>
+      <string>#{var}</string>
```
Comment (JCount) [inline Formula/mariadb@10.1.rb]: There is the same question here about conflicts in the `ProgramArguments` and `WorkingDirectory` keys.

File: Formula/mariadb.rb (hunk 3 above)
Comment (makigumo) [inline Formula/mariadb.rb]: Well 10.1.23 got released after 10.2.5 and contains fixes that were not yet in 10.2.5. :)

**Review by JCount — DISMISSED** [review]

File: Formula/mariadb@10.1.rb (hunk 5 above)
Comment (makigumo) [inline Formula/mariadb@10.1.rb]: Currently it seems it is left to the user to take care of when looking at similar formulae e.g. `mysql@5.5` etc.

File: Formula/mariadb@10.1.rb (hunk 2 above)
Comment (makigumo) [inline Formula/mariadb@10.1.rb]: Done.

File: Formula/mariadb@10.1.rb (hunk 1 above)
Comment (makigumo) [inline Formula/mariadb@10.1.rb]: Removed.

File: Formula/mariadb@10.1.rb [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    server starting up correctly.
+
+    MySQL is configured to only allow connections from localhost by default
+
+    To connect:
+        mysql -uroot
+    EOS
+  end
+
+  plist_options :manual => "mysql.server start"
```
Comment (JCount) [inline Formula/mariadb@10.1.rb]: This needs to be `plist_options :manual => "#{HOMEBREW_PREFIX}/opt/mariadb@10.1/bin/mysql.server start"`.

Comment (makigumo) [inline Formula/mariadb@10.1.rb]: Done.

File: Formula/mariadb@10.1.rb [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    # Link the setup script into bin
+    bin.install_symlink prefix/"scripts/mysql_install_db"
+
+    # Fix up the control script and link into bin
+    inreplace "#{prefix}/support-files/mysql.server", /^(PATH=".*)(")/, "\\1:#{HOMEBREW_PREFIX}/bin\\2"
+
+    bin.install_symlink prefix/"support-files/mysql.server"
+
+    # Move sourced non-executable out of bin into libexec
+    libexec.mkpath
```
Comment (JCount) [inline Formula/mariadb@10.1.rb]: This line is not needed. (Sorry for pointing these things out one by one, I keep missing things.)

Comment (makigumo) [inline Formula/mariadb@10.1.rb]: Removed.

**Review by ilovezfs — APPROVED** [review]

**Review by JCount — APPROVED** [review]

**Comment by JCount** [comment]
@makigumo Thank you for your contribution to Homebrew! 🎆

**Pull request closed without merging**